### ISSUE TITLE
Zhang implementation

### ISSUE THREAD

**Issue by thomkeh** [predictive-analytics-lab/EthicML#291]
Oliver:

> I feel as though implementing zhang’s mitigating unwanted bias by Adversarial learning algorithm is possible to implement in a way that doesn’t suck. In his original paper he hand writes a den layer network and in aif 360 they do the same. If there’s not a way in tensorflow or pytorch to easily the project the gradients with respect to one loss onto the gradients with respect to another loss perhaps we’ll have to go a bit lower level. It might be a good opportunity to play around with JAX https://colab.research.google.com/github/google/jax/blob/master/notebooks/neural_network_and_data_loading.ipynb

**Comment by olliethomas** [comment]
Has this been added?

**Comment by thomkeh** [comment]
I think Myles considers this to be the implementation: #293 . But it's kind of only one part.
